### PR TITLE
fix: validate manifest file and improve error handling in _load_manifest_from_file

### LINKED ISSUE #2817: Raise a clear error when the dbt manifest is empty or invalid (LoadMode.DBT_MANIFEST)
### Description

When using `LoadMode.DBT_MANIFEST`, Cosmos loads the manifest in `DbtGraph.load_from_dbt_manifest` via `json.load(fp)` on the configured `manifest_path`. If the file is empty, truncated, or not valid JSON, `json.load` raises a raw `json.JSONDecodeError`. Because this runs at DAG-parse time (inside `DbtTaskGroup`/`DbtDag` construction), the unhandled exception **breaks DAG import** and surfaces only as an opaque `JSONDecodeError` deep in Cosmos internals — with no indication of which manifest path failed or that the file is malformed.

### Current behavior

A malformed/empty `manifest.json` fails with a bare traceback through `cosmos/dbt/graph.py` → `load_from_dbt_manifest` → `json.load`:

```
File ".../cosmos/airflow/task_group.py", line ..., in __init__
File ".../cosmos/converter.py", line ..., in __init__
    self.dbt_graph.load(method=..., execution_mode=...)
File ".../cosmos/dbt/graph.py", line ..., in load_from_dbt_manifest
    manifest = json.load(fp)
...
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

The error gives no hint about the cause or which path was read.

### Expected behavior

Validate the manifest before parsing and raise an actionable `CosmosLoadDbtException` that includes the path and the underlying cause, e.g.:

```
Failed to load dbt manifest at <path>: file is empty or not valid JSON (<original error>)
```

### Proposed implementation

* In `load_from_dbt_manifest`, check the file exists / is non-empty and wrap the `json.load` call to re-raise as `CosmosLoadDbtException` with the manifest path and original error.
* Add a test with an empty/truncated manifest fixture.

### Notes

Related parsing work: astronomer/astronomer-cosmos#2294, astronomer/astronomer-cosmos#2486. This is a robustness / error-message improvement on the `LoadMode.DBT_MANIFEST` path (not a performance change).

### PR DESCRIPTION
**Pull request by April-Sonnet**
Improve error handling when loading a dbt manifest JSON file. Previously, attempting to load an empty or invalid manifest file would raise an opaque JSONDecodeError with no indication of which file failed or why.

Changes:
1. **Empty file check**: Before parsing, verify the file is non-empty and raise CosmosLoadDbtException with the path
2. **JSON decode error handling**: Wrap json.load/orjson.loads in try/except for JSONDecodeError/ValueError
3. **File stat error handling**: Wrap stat() call to catch OSError

Closes #2817

### CODE REVIEW

**Review by pankajastro — COMMENTED** [review]
Thanks for the PR. `stat()` adds a separate round-trip (and the `ObjectStoragePath.stat().st_size` ahead of the read we do anyway — since we read the bytes regardless, drop it and check if not content: after reading once. Could you also add empty-file / invalid-JSON tests in test_orjson_parser.py?

```python3
try:
      with manifest_path.open("rb") as fp:
          content = fp.read()
  except OSError as e:
      raise CosmosLoadDbtException(f"Failed to load dbt manifest at `{manifest_path}`: {e}") from e

  if not content:
      raise CosmosLoadDbtException(f"Failed to load dbt manifest at `{manifest_path}`: file is empty")
  
  try:
      manifest = orjson.loads(content) if (settings.enable_orjson_parser and orjson) else json.loads(content)
  except (json.JSONDecodeError, ValueError) as e:
      raise CosmosLoadDbtException(f"... file is not valid JSON ({e})") from e

```

File: cosmos/dbt/graph.py [hunk 1]
```diff
@@ -1356,16 +1356,24 @@ def _load_manifest_from_file(self, manifest_path: Path | ObjectStoragePath) -> d
...
-                manifest = json.load(fp)
+                content = fp.read()
+        except OSError as e:
+            raise CosmosLoadDbtException(f"Failed to load dbt manifest at `{manifest_path}`: {e}") from e
+
+        if not content:
+            raise CosmosLoadDbtException(f"Failed to load dbt manifest at `{manifest_path}`: file is empty")
+
+        try:
+            manifest = orjson.loads(content) if (settings.enable_orjson_parser and orjson) else json.loads(content)
```
Comment (tatiana) [inline cosmos/dbt/graph.py:1372]: When `enable_orjson_parser=True` but orjson is unavailable, it silently falls back to stdlib json instead of raising the existing “pip install orjson” `CosmosLoadDbtException`

**Review by tatiana — COMMENTED** [review]
@April-Sonnet Thanks for your interest in contributing to Cosmos. I wanted to share some feedback that I hope will be useful going forward.

AI tools can absolutely help contributors work more effectively. That said, they should improve the quality of contributions and not add review overhead or create extra work for maintainers and the community.

Open source is fundamentally about collaborative improvement. Contributions should come from actually understanding the problem and working toward a meaningful solution, rather than hunting for "good first issue" labels and using automated tools to generate fixes with minimal engagement.

In this case, you opened two separate PRs for the same issue:

#2818
#2829
Opening multiple PRs for the same issue before resolving the existing one creates unnecessary duplication and makes things harder for reviewers. If you decide to work on something, please focus on a single PR and iterate based on feedback.

Also, both PRs currently have failing checks. That suggests the changes weren't validated before submission. Please make sure CI passes and the solution has been properly tested before opening a PR.

Because of all this, I'm closing both PRs. Another contributor already submitted a solution in #2819 that shows a clearer understanding of the problem and a more complete implementation.

We'd still welcome future contributions, but please take the time to understand the issue, validate your changes, and keep in mind the review effort required of maintainers and other contributors.

Some open source projects are [starting to limit community contributions](https://ladybird.org/posts/changing-how-we-develop-ladybird/) due to similar patterns we're seeing. It's a real shame, and I don't want us to end up having to do the same with Cosmos.

**Review by tatiana — COMMENTED** [review]
@April-Sonnet Thanks for your interest in contributing to Cosmos. I wanted to share some feedback that I hope will be useful going forward.

AI tools can absolutely help contributors work more effectively. That said, they should improve the quality of contributions and not add review overhead or create extra work for maintainers and the community.

Open source is fundamentally about collaborative improvement. Contributions should come from actually understanding the problem and working toward a meaningful solution, rather than hunting for "good first issue" labels and using automated tools to generate fixes with minimal engagement.

In this case, you opened two separate PRs for the same issue:
- #2818
- #2829

Opening multiple PRs for the same issue before resolving the existing one creates unnecessary duplication and makes things harder for reviewers. If you decide to work on something, please focus on a single PR and iterate based on feedback.

Also, both PRs currently have failing checks. That suggests the changes weren't validated before submission. Please make sure CI passes and the solution has been properly tested before opening a PR.

Because of all this, I'm closing both PRs. Another contributor already submitted a solution in #2819 that shows a clearer understanding of the problem and a more complete implementation.

We'd still like to welcome future contributions, but please take time to understand the issue, validate your changes, and keep in mind the review effort required from maintainers and other contributors.

Some open source projects are [starting to limit community contributions](https://ladybird.org/posts/changing-how-we-develop-ladybird/) due to similar patterns we're seeing. It's a real shame, and I don't want us to end up having to do the same with Cosmos.
